### PR TITLE
[ORCA-3465] - Event Orchestration Service path resource

### PR DESCRIPTION
**Pull request by alenapan**
### Changes
- Add support for the `pagerduty_event_orchestration_service` resource
- Modify the unrouted resource to use shared method for variables and extractions

### Testing
image

### CODE REVIEW

File: pagerduty/event_orchestration_path_util.go [hunk 1]
```diff
@@ -25,3 +28,105 @@ var PagerDutyEventOrchestrationPathConditions = map[string]*schema.Schema{
 		Required: true,
 	},
 }
+
+func checkExtractions(context context.Context, diff *schema.ResourceDiff, i interface{}) error {
```
Comment (alenapan) [inline pagerduty/event_orchestration_path_util.go:32]: this is used to validate extraction properties

File: pagerduty/resource_pagerduty_event_orchestration_path_service.go [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+		Type:     schema.TypeString,
+		Required: true,
+	},
+	"value": {
+		Type:     schema.TypeString,
+		Required: true,
+	},
+}
+
+var eventOrchestrationPathServiceCatchAllActions = map[string]*schema.Schema{
```
Comment (alenapan) [inline pagerduty/resource_pagerduty_event_orchestration_path_service.go:23]: the only difference between catch_all and regular rule actions is the `route_to` attribute, that's why I extracted the schema snippet to a reusable variable

File: pagerduty/util.go [hunk 3]
```diff
@@ -145,3 +145,10 @@ func stringPtrToStringType(v *string) string {
 	}
 	return *v
 }
+
+func intTypeToIntPtr(v int) *int {
```
Comment (alenapan) [inline pagerduty/util.go:149]: used for `Suspend`

File: pagerduty/resource_pagerduty_event_orchestration_path_service.go [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+	for _, rule := range items {
+		r := rule.(map[string]interface{})
+
+		ruleInSet := &pagerduty.EventOrchestrationPathRule{
+			ID:       r["id"].(string),
+			Label:    r["label"].(string),
+			Disabled: r["disabled"].(bool),
+			// TODO: move conditions logic to util
+			Conditions: expandRouterConditions(r["conditions"].(interface{})),
+			Actions:    expandServicePathActions(r["actions"].([]interface{})),
```
Comment (alexzakabluk) [inline pagerduty/resource_pagerduty_event_orchestration_path_service.go:380]:
```suggestion
			Conditions: expandRouterConditions(r["conditions"]),
```

File: pagerduty/resource_pagerduty_event_orchestration_path_service.go [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+
+	for _, rule := range items {
+		r := rule.(map[string]interface{})
+
+		ruleInSet := &pagerduty.EventOrchestrationPathRule{
+			ID:       r["id"].(string),
+			Label:    r["label"].(string),
+			Disabled: r["disabled"].(bool),
+			// TODO: move conditions logic to util
+			Conditions: expandRouterConditions(r["conditions"].(interface{})),
```
Comment (alexzakabluk) [inline pagerduty/resource_pagerduty_event_orchestration_path_service.go:379]:
```suggestion
			Conditions: expandRouterConditions(r["conditions"]),
```

File: pagerduty/resource_pagerduty_event_orchestration_path_service_test.go [hunk 6]
```diff
@@ -0,0 +1,793 @@
...
+
+func init() {
+	resource.AddTestSweepers("pagerduty_event_orchestration_service", &resource.Sweeper{
+		Name: "pagerduty_event_orchestration_service",
+		F:    testSweepEventOrchestration,
+		Dependencies: []string{
+			"pagerduty_user",
+			"pagerduty_escalation_policy",
+			"pagerduty_service",
+		},
```
Comment (alexzakabluk) [inline pagerduty/resource_pagerduty_event_orchestration_path_service_test.go:22]:
```suggestion
```

File: pagerduty/resource_pagerduty_event_orchestration_path_service_test.go [hunk 7]
```diff
@@ -0,0 +1,793 @@
...
+		client, _ := testAccProvider.Meta().(*Config).Client()
+		found, _, err := client.EventOrchestrationPaths.Get(orch.Primary.ID, "service")
+		if err != nil {
+			return err
+		}
+		if found.Parent.ID != orch.Primary.ID {
+			return fmt.Errorf("Event Orchrestration Service not found: %v - %v", orch.Primary.ID, found)
+		}
+
+		// return fmt.Errorf(">>> attr: %v", orch.Primary.Attributes)
```
Comment (alexzakabluk) [inline pagerduty/resource_pagerduty_event_orchestration_path_service_test.go:216]:
```suggestion
```

**Review by alexzakabluk — APPROVED** [review]

**Pull request merged**